### PR TITLE
Implement enable/disable 'designated_gateway' feature in aviatrix_gateway

### PR DESCRIPTION
**Pull request by zhiqiangjin**

### CODE REVIEW

**Comment by zhiqiangjin** [comment]
Fixes #37

File: aviatrix/resource_aviatrix_gateway.go [hunk 1]
```diff
@@ -719,6 +754,14 @@ func resourceAviatrixGatewayRead(d *schema.ResourceData, meta interface{}) error
 			d.Set("allocate_new_eip", true)
 		}
 
+		if gw.CloudType == 1 && (gw.EnableDesignatedGateway == "Yes" || gw.EnableDesignatedGateway == "yes") {
```
Comment (arvindsree) [inline aviatrix/resource_aviatrix_gateway.go:757]: Not sure if you need the gateway cloud type check, you can directly set it

File: goaviatrix/gateway.go [hunk 2]
```diff
@@ -689,3 +691,32 @@ func (c *Client) DisableVpnNat(gateway *Gateway) error {
 	}
 	return nil
 }
+
+func (c *Client) EditDesignatedGateway(gateway *Gateway) error {
+	Url, err := url.Parse(c.baseURL)
+	if err != nil {
+		return errors.New(("url Parsing failed for 'set_designated_gateway_additional_cidr_list'") + err.Error())
+	}
+	disableVpnNat := url.Values{}
```
Comment (arvindsree) [inline goaviatrix/gateway.go:700]: change to meaningful name like EditDesignatedGateway?

**Review by arvindsree — APPROVED** [review]

File: aviatrix/resource_aviatrix_gateway.go (hunk 1 above)
Comment (zhiqiangjin) [reply aviatrix/resource_aviatrix_gateway.go:757]: Yes, can directly set it.

**Pull request merged**